test/integration/test_job.py

```diff
@@ -122,25 +122,20 @@ def test_run_program_failed_ran_too_long(self, service):
     def test_run_program_failed_custom_max_execution_time(self, service):
         """Test a program that failed with a custom max_execution_time."""
         # check that program max execution time is overridden
-        max_execution_time = 300
+        program_max_execution_time = 400
+        job_max_execution_time = 350
         program_id = self._upload_program(
-            service, max_execution_time=max_execution_time
+            service, max_execution_time=program_max_execution_time
         )
-        inputs = {"iterations": 1, "sleep_per_iteration": 60}
         job = self._run_program(
-            service, program_id=program_id, inputs=inputs, max_execution_time=1
+            service, program_id=program_id, max_execution_time=job_max_execution_time
         )
         job.wait_for_final_state()
-        job_result_raw = service._api_client.job_results(job.job_id)
-        self.assertEqual(JobStatus.ERROR, job.status())
-        self.assertIn(
-            API_TO_JOB_ERROR_MESSAGE["CANCELLED - RAN TOO LONG"].format(
-                job.job_id, job_result_raw
-            ),
-            job.error_message(),
+        self.assertEqual(
+            job._api_client.job_get(job.job_id)["cost"], job_max_execution_time
         )
-        with self.assertRaises(RuntimeJobFailureError):
-            job.result()
+        with self.assertRaises(IBMRuntimeError):
+            self._run_program(service, program_id=program_id, max_execution_time=1)
```

Comment:
I think it's better if we split this into 2 separate tests, one to check if the job cost is set and the other to test if it fails when atleast a minimum of 300 is not given...